api/api.go

```diff
@@ -0,0 +1,106 @@
+/*
+management-interface - Web based management of Raspberry Pis over WiFi
+Copyright (C) 2018, The Cacophony Project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package managementapi
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	cptvDir  = "/var/spool/cptv/"
+	cptvGlob = "*.cptv"
+)
+
+// GetRecordings returns a list of cptv files in a array.
+func GetRecordings(w http.ResponseWriter, r *http.Request) {
+	log.Println("get recordings")
+	names := getCptvNames()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(names)
+}
+
+// GetRecording downloads a cptv file
+func GetRecording(w http.ResponseWriter, r *http.Request) {
+	recordingName := mux.Vars(r)["id"]
+	log.Printf("get recording '%s'", recordingName)
+	if !checkIfCptvFile(recordingName) {
+		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, "cptv file not found\n")
+		return
+	}
+
+	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", recordingName))
+	w.Header().Set("Content-Type", "application/x-cptv")
+	f, err := os.Open(filepath.Join(cptvDir, recordingName))
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Println(err)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	reader := bufio.NewReader(f)
+	io.Copy(w, reader)
+}
+
+// DeleteRecording deletes the given cptv file
+func DeleteRecording(w http.ResponseWriter, r *http.Request) {
+	cptvName := mux.Vars(r)["id"]
+	log.Printf("delete cptv '%s'", cptvName)
+	if !checkIfCptvFile(cptvName) {
+		w.WriteHeader(http.StatusOK)
+		io.WriteString(w, "cptv file not found\n")
+		return
+	}
+	err := os.Remove(filepath.Join(cptvDir, cptvName))
+	if err != nil {
```

Comment:
To guard against a file being removed in between the `checkIfCptvFile` call and here, use `os.IsNotExist(err)`. If that returns true, treat this the same as success.

Reply:
so basically, `if err != nil && !os.IsNotExist(err) {`